src/simplify_algebra.cpp

```diff
@@ -57,6 +57,22 @@ auto conv_const_weights()
         match::args(match::none_of(match::is_constant()), match::is_constant().bind("w")));
 }
 
+auto from_int4()
+{
+    return match::make_predicate_matcher([](instruction_ref start) {
+        return fix<bool>([&](auto self, instruction_ref ins) {
+            auto alias = instruction::get_output_alias(ins);
+            if(contains({"reshape", "dequantizelinear"}, alias->name()))
+                return self(alias->inputs().front());
+            if(alias->name() == "concat")
+                return all_of(alias->inputs(), self);
```

Comment:
Shouldn't this be `any_of`? Also, the name of this function/lambda `from_int4` should say `any` or `all`, just to make its intent clear. Thanks.

Reply:
> Shouldn't this be any_of? 

No, if its mixed then we cant get int4 out of it.